### PR TITLE
Reusable normal button#23

### PR DESCRIPTION
**Pull request by kevinkkimutai**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reference the issue related to this PR as shown below. Every issue has it's own number you can find the issue numbers [here](https://github.com/italanta/italanta-apps/issues).

Fixes # (issue) - *Example*: Fixes #10, #20 NOTE: GitHub issue numbers should be prefixed with **#** to aid with issue tracking.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
-[x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

-[x] Test A
- [x] Test B

**Test Configuration**: (optional)

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [x] My code follows the style guidelines of this project
-  [x]  I have performed a self-review of my code
-  [x]  I have commented my code, particularly in hard-to-understand areas
-  [x]  I have made corresponding changes to the documentation
-  [x]  My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
-  [x]  New and existing unit tests pass locally with my changes
-  [x]  Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'elewa-website-elewa-normal-button',
+  templateUrl: './elewa-normal-button.component.html',
+  styleUrls: ['./elewa-normal-button.component.scss'],
+})
+/*Pass data to the component */
+export class ElewaNormalButtonComponent {
+  @Input() buttonData!: ButtonData;
+
+  @Output() buttonClick: EventEmitter<void> =new EventEmitter<void>
+  ();
```
Comment (LemmyMwaura) [inline libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts:14]: move the function call to the same line `new Event...();`

File: libs/models/schema/ui/buttons/src/lib/models-schema-ui-buttons.module.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { NgModule } from '@angular/core';
```
Comment (LemmyMwaura) [inline libs/models/schema/ui/buttons/src/lib/models-schema-ui-buttons.module.ts:1]: There's no need for this file. The module has no declarations to manage

File: nx.json [hunk 3]
```diff
@@ -6,29 +6,53 @@
...
+      "runner": "nx-cloud",
       "options": {
-        "cacheableOperations": ["build", "lint", "test", "e2e"]
+        "cacheableOperations": [
+          "build",
+          "lint",
+          "test",
+          "e2e"
+        ],
+        "accessToken": "YjMzZDc4ZTItZWY1Ni00NmQ2LTkxZjMtMmIyYWJiNGNiMTlkfHJlYWQtd3JpdGU="
```
Comment (LemmyMwaura) [inline nx.json:17]: There's no need for this change.

File: package.json [hunk 4]
```diff
@@ -68,6 +68,7 @@
     "jest-environment-node": "^29.4.1",
     "jest-preset-angular": "13.1.1",
     "nx": "16.6.0",
+    "nx-cloud": "^16.4.0-beta.1",
```
Comment (LemmyMwaura) [inline package.json:71]: No need for you to use nx-cloud

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
A few changes left

File: libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts (hunk 1 above)
Comment (kevinkkimutai) [reply libs/elements/layout/buttons/src/lib/components/elewa-normal-button/elewa-normal-button.component.ts:14]: changes made

File: libs/models/schema/ui/buttons/src/lib/models-schema-ui-buttons.module.ts (hunk 2 above)
Comment (kevinkkimutai) [reply libs/models/schema/ui/buttons/src/lib/models-schema-ui-buttons.module.ts:1]: file removed

File: nx.json (hunk 3 above)
Comment (kevinkkimutai) [reply nx.json:17]: made changes

File: package.json (hunk 4 above)
Comment (kevinkkimutai) [reply package.json:71]: nx-cloud removed

**Comment by LemmyMwaura** [comment]
I'm closing this as it's been resolved in #72, but good job implementing some of the requested changes. A few things were missing from your PR. You can check the merged version as a reference point.

**Pull request closed without merging**